Sources/NIOHTTPClient/HTTPHandler.swift

```
@@ -442,6 +442,8 @@ internal struct RedirectHandler<T> {
 
         var request = self.request
         request.url = redirectURL
+        request.host = redirectURL.host!
```

host and scheme can be safely unwrapped as they have already been validated earlier

---

imo we should be defensive here, so we don't make assumption about the call site which could change over time

---

@tomerd I was thinking that maybe host and scheme should be resolved on bootstrap connect, that way host and scheme could be optional and there would be no need to have throwing `HTTPClient.Request` init? Most helper methods accept string as an URL, so error will be inside returned future anyway... What do you think?

---

+1 on non-throwing `HTTPClient.Request` ctor

---

this should probably be done in separate PR to reduce scope, maybe create an issue?

---

@artemredkin @tomerd Does this mean I can leave this as it is for now? If yes, I have fixed all reviews and the PR should now be mergeable.

---

@adtrevor how about you add a check here until @artemredkin takes on the bigger change? can be a simple as 

```
if let host = redirectURL.host {
  request.host = host
}
```

---

@tomerd Thanks! Done and added an `assertionFailure` on else so that it doesn't silently fail on debug as it's still something unexpected.

---

Thank you, LGTM! One small comment though, I think `preconditionFailure` would be more appropriate here, imo (we use it in other parts of the codebase as well).

---

@artemredkin Replaced it! Thanks!